src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/component/CommunityAPIClient.kt

```diff
@@ -51,22 +58,51 @@ class CommunityAPIClient(
       .bodyToMono(responseBodyClass)
       .onErrorMap { e ->
         handleResponse(e, "")
-        e
       }
       .block()
   }
 
-  fun handleResponse(e: Throwable, requestBody: Any): String {
+  fun handleResponse(e: Throwable, requestBody: Any): DownstreamApiCallError {
     val responseBodyAsString = when (e) {
       is BadRequest -> e.responseBodyAsString
       else -> e.localizedMessage
     }
-    logger.error(
-      "Call to community api failed",
-      e,
-      StructuredArguments.kv("req.body", requestBody),
-      StructuredArguments.kv("res.body", responseBodyAsString)
-    )
-    return responseBodyAsString.let { it } ?: "No response body message"
+
+    val statusCode = when (e) {
+      is WebClientResponseException -> e.statusCode
+      else -> INTERNAL_SERVER_ERROR
+    }
+
+    val causeMessage = userMessageOrDeveloperMessageOrResponseBodyInThatOrder(responseBodyAsString)
+    val error = DownstreamApiCallError(statusCode, causeMessage, responseBodyAsString, e)
+    if (error.logError) {
+      logger.error(
```

Comment:
Feels like this should now be done in `@ExceptionHandler(DownstreamApiCallError::class)`

Reply:
Made it unconditional